server/api/crud/pipelines.py

```diff
@@ -90,8 +90,8 @@ def list_pipelines(
                 page_token=page_token,
                 page_size=page_size
                 or mlrun.common.schemas.PipelinesPagination.default_page_size,
-                sort_by=sort_by,
-                filter=filter_,
+                sort_by=sort_by if page_token == "" else "",
+                filter=filter_ if page_token == "" else "",
```

Comment:
These are not needed, since in this flow (querying across projects with `*` for project name) the BE does not do any iteration internally, but rather just passes the parameters that the user provided to kfp. In this case, the responsibility for the validity of the params falls on the users and we should not make decisions for them.

I don't think there's anything wrong with doing this, but TBH I'm not sure what my opinion on this is. After all, even putting a warning here won't really matter much since the user will not see it (it will be in the MLRun service logs).

Reply:
I guess you're right, we can let the call fail because we don't iterate by ourselves, so they user will get an error status code.
But, will the `list_runs` call raise an exception in such case?

Reply:
Yes, since kfp will raise an exception (you can try to run it on a setup without your fix and see). I do suspect that the error will not be extremely visible to the user... Trying to do something on my setup, looks like it's returning a 500 with 
```
{
  "detail": "ApiException()"
}
```
That is not very helpful.
Given this, it's probably better to either make this succeed (keep your validation) or return a specific error message that explained what happened.

Reply:
In that case, I might go with a try-except and wrap the exception in a more informative error